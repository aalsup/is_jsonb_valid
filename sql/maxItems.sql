-- maxItems validation
-- shorter is valid
SELECT is_jsonb_valid('{"maxItems":2}', '[1]');
-- exact length is valid
SELECT is_jsonb_valid('{"maxItems":2}', '[1,2]');
-- too long is invalid
SELECT is_jsonb_valid('{"maxItems":2}', '[1,2,3]');
-- ignores non-arrays
SELECT is_jsonb_valid('{"maxItems":2}', '"foobar"');