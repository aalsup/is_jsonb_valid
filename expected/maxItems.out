-- maxItems validation
-- shorter is valid
SELECT is_jsonb_valid('{"maxItems":2}', '[1]');
 is_jsonb_valid 
----------------
 t
(1 row)

-- exact length is valid
SELECT is_jsonb_valid('{"maxItems":2}', '[1,2]');
 is_jsonb_valid 
----------------
 t
(1 row)

-- too long is invalid
SELECT is_jsonb_valid('{"maxItems":2}', '[1,2,3]');
 is_jsonb_valid 
----------------
 f
(1 row)

-- ignores non-arrays
SELECT is_jsonb_valid('{"maxItems":2}', '"foobar"');
 is_jsonb_valid 
----------------
 t
(1 row)

